x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/SparseFileTracker.java

```diff
@@ -358,9 +432,13 @@ public String toString() {
         long end;
 
         @Nullable // if not pending
-        final PlainListenableActionFuture<Void> completionListener;
+        final ProgressListenableActionFuture completionListener;
+
+        Range(long start, long end) {
```

Comment:
I expected this constructor to make a _completed_ range whereas in fact we use the other one and pass `null` as the listener. There's only a couple of call-sites, I'd prefer to inline this to avoid that confusion.